core/templates/dev/head/components/HintAndSolutionButtonsDirective.js

```diff
@@ -0,0 +1,95 @@
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/hint_and_solution_buttons_directive.html'),
+      controller: [
+        '$scope', '$rootScope', 'HintManagerService',
+        'ExplorationPlayerService', 'SolutionManagerService',
+        'PlayerTranscriptService', 'HintAndSolutionModalService',
+        'DeviceInfoService', 'WindowDimensionsService',
+        function(
+          $scope, $rootScope, HintManagerService,
+          ExplorationPlayerService, SolutionManagerService,
+          PlayerTranscriptService, HintAndSolutionModalService,
+          DeviceInfoService, WindowDimensionsService) {
+          $scope.defaultHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_hint.svg');
+
+          $scope.activeHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.defaultSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_solution.svg');
+
+          $scope.activeSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.hintPositions = [];
+          $scope.solutionExists = null;
+          $scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow();
+          $scope.activeModalIndex = null;
+
+          $rootScope.$on('hintsAndSolutionReset', function(evt, data) {
+            for(var index = 0; index < data.numOfHints; index++) {
+              $scope.hintPositions.push(index);
+            }
+            $scope.solutionExists = data.solutionExists;
+          });
+
+          $scope.isHintButtonVisibleAtIndex = function(index) {
+            if (index === 0) {
+              return HintManagerService.isCurrentHintAvailable() ||
+                index < HintManagerService.getCurrentHintIndex();
+            } else if (index === HintManagerService.getCurrentHintIndex()) {
+              return HintManagerService.isCurrentHintAvailable();
+            } else {
+              return index < HintManagerService.getCurrentHintIndex();
+            }
+          };
+
+          $scope.isSolutionButtonVisible = function() {
+            return SolutionManagerService.isCurrentSolutionAvailable() &&
+                HintManagerService.areAllHintsExhausted();
+          };
+
+          $scope.displayHintModalForIndex = function(index) {
+            $scope.activeModalIndex = index;
+            var promise = (
+              HintAndSolutionModalService.displayHintModalForIndex(index));
+            promise.result.then(null, function() {
+              $scope.activeModalIndex = null;
+            });
+          };
+
+          $scope.displaySolutionModal = function() {
+            $scope.activeModalIndex = -1;
+            HintAndSolutionModalService.displaySolutionModal().result.then(null,
```

Comment:
Might as well use the same style for this and the function above? I.e. declare a `var promise` for both, or don't declare it for both.